src/libslic3r/PrintConfig.cpp

```diff
@@ -3361,6 +3361,11 @@ std::string FullPrintConfig::validate()
         switch (opt->type()) {
         case coFloat:
         case coPercent:
+        {
+            auto* fopt = static_cast<const ConfigOptionPercent*>(opt);
+            out_of_range = fopt->get_abs_value(100) < optdef->min || fopt->get_abs_value(100) > optdef->max;
+            break;
+        }
```

Comment:
Thanks for your work. But can I ask you to explain what exactly your commit does? FullPrintConfig::validate() makes validation of the values contained  in current configuration.
And this part of code 
```
 case coFloat:
 case coPercent:
 case coFloatOrPercent:
 {
     auto *fopt = static_cast<const ConfigOptionFloat*>(opt);
     out_of_range = fopt->value < optdef->min || fopt->value > optdef->max;
     break;
}
```
validates fopt->value for out of range [optdef->min, optdef->max] for options with type coFloat, coPercent and coFloatOrPercent. And it is exactly what we want.
But I can't understand what are you trying to achieve here.

Reply:
It's a bit old, and it was not good.

basically, currently, the code check for "value" vs min / max, but percentage are  *100 vs inteneded, but the min/max check doesn't takes that into account.

What I do now is: 
`fopt->value < optdef->min` for coFloat / coFloats
`fopt->get_abs_value(100) < optdef->min` for coPercent / coPercents
`fopt->get_abs_value(1) < optdef->min` for coFloatOrPercent